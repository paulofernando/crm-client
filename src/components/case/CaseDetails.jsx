import React from 'react';
import NumberFormat from 'react-number-format';
import Moment from 'react-moment';

import '../ComponentStyle.css';
import ContactTile from "../contact/ContactDetails";

const CourtCaseDetails = (props) => { 
    return (
        <div className="courtCaseCard courtCaseDetailed">
            <h3 className="courtCaseTitle">{props.courtCase.title}</h3>
            <p>{props.courtCase.description}</p>
            <div className="cardContainer">
                {props.courtCase.contacts &&
                    props.courtCase.contacts.map(contact => (
                        <ContactTile key={contact.id} contact={contact} />
                    ))}
            </div>
            <Moment format="MMMM Do YYYY">
                {props.courtCase.courtDate}
            </Moment>
            <div className="courtCaseValue">
                <NumberFormat value={props.courtCase.value} displayType={'text'} thousandSeparator={true} prefix={'$'} />
            </div>
        </div>
    );    
}

export default CourtCaseDetails;