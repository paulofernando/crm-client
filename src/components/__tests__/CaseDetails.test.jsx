import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import CaseDetails from '../case/CaseDetails';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

// -------------------------------

const courtCase = {
    title: "Test title",
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin arcu enim, tincidunt fringilla rhoncus vitae, semper quis quam. Nam et tincidunt ex. Nunc ut ante id lorem consectetur vulputate ut non nulla. Curabitur viverra, libero vel rutrum volutpat, erat orci eleifend sem, sit amet vestibulum leo lectus at quam",
    value: 1000.00,
    courtDate: "2020-01-01T00:00:00.000Z"
}

const CaseDetailsComponent = (<CaseDetails courtCase={courtCase} />)

// -------------------------------

describe('CaseDetails', () => {
    it('renders correctly', () => {
        const wrapper = renderer
            .create(CaseDetailsComponent)
            .toJSON();
        expect(wrapper).toMatchSnapshot();

        const component = mount(
            <CaseDetails courtCase={courtCase} />
        );

        const courtCaseTitle = component.find('.courtCaseTitle');
        expect(courtCaseTitle.text()).toBe('Test title');

        const courtCaseDescription = component.find('.courtCaseDescription');
        expect(courtCaseDescription.text()).toBe('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin arcu enim, tincidunt fringilla rhoncus vitae, semper quis quam. Nam et tincidunt ex. Nunc ut ante id lorem consectetur vulputate ut non nulla. Curabitur viverra, libero vel rutrum volutpat, erat orci eleifend sem, sit amet vestibulum leo lectus at quam');

        const courtCaseDate = component.find('.courtCaseDate');
        expect(courtCaseDate.text()).toBe('2020-01-01T00:00:00.000Z');

        const courtCaseValue = component.find('.courtCaseValue');
        expect(courtCaseValue.text()).toBe('$1000.00');        
    });
});