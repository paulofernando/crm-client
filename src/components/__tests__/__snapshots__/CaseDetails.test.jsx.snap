// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseDetails renders correctly 1`] = `
<div
  className="courtCaseCard courtCaseDetailed"
>
  <h3
    className="courtCaseTitle"
  >
    Test title
  </h3>
  <p
    className="courtCaseDescription"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin arcu enim, tincidunt fringilla rhoncus vitae, semper quis quam. Nam et tincidunt ex. Nunc ut ante id lorem consectetur vulputate ut non nulla. Curabitur viverra, libero vel rutrum volutpat, erat orci eleifend sem, sit amet vestibulum leo lectus at quam
  </p>
  <div
    className="cardContainer"
  />
  <time
    className="courtCaseDate"
    dateTime={"2020-01-01T00:00:00.000Z"}
  >
    December 31st 2019
  </time>
  <span
    className="courtCaseValue"
  >
    $1,000
  </span>
</div>
`;
